lib/1.4/dml-builtins.dml

```diff
@@ -532,15 +532,24 @@ template device {
 
 Group objects are generic container objects, used to group other objects.
 They can appear anywhere in the object hierarchy, but some object types
-(currently `implement`, `interface`, `bank` and `port`) may not have a group
+(currently `implement` and `interface`) may not have a group
 as parent.
 
 The `group` template contains no particular methods or parameters other
 than what is inherited from the [`object`](#object) template.
+
+In a group hierarchy directly below a `port` or `bank` object, none of the
+group object may be named `device` or `port`; this is to avoid name clashes.
 */
 template group is object {
     param objtype = "group";
     param log_group = undefined;
+    #if ((_nongroup_parent.objtype == "port"
+          || _nongroup_parent.objtype == "device")
+         && (this.name == "port" || this.name == "bank")) {
```

Comment:
This check contradicts the documentation: this forbids groups from being named "port" or "bank" if the group's a part of a `port` or `device`.

Reply:
docs fixed